### PR TITLE
[new release] equinoxe (0.1.0)

### PR DESCRIPTION
**Pull request by maiste**
An OCaml wrapper for the Equinix API

- Project page: <a href="https://github.com/maiste/equinoxe">https://github.com/maiste/equinoxe</a>
- Documentation: <a href="maiste.github.io/equinoxe">maiste.github.io/equinoxe</a>

### Added

- Import a default http client to communicate with the Equinix API (maiste/equinoxe#25, maiste/equinoxe#22, maiste/equinoxe#23, @maiste)
- Add some user commands (maiste/equinoxe#24, @maiste)
- Add some organization commands (maiste/equinoxe#33, @maiste)
- Add some projects commands (maiste/equinoxe#36, maiste/equinoxe#37, @maiste)
- Add some devices commands (maiste/equinoxe#37, @maiste)
- JSON manager module (maiste/equinoxe#6, @maiste)
- IP module (maiste/equinoxe#47, @maiste)
- Generate a test system (maiste/equinoxe#42, @maiste)
- Add a README (maiste/equinoxe#26, maiste/equinoxe#27, @maiste)

### CODE REVIEW

**Comment by maiste** [comment]
This is a continuation of  #20332. It was better to open a new one as the branch was updated a lot.

File: packages/equinoxe/equinoxe.0.1.0/opam [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  "dune" {>= "2.9"}
+  "ocaml" {>= "4.08.0"}
+  "odoc" {with-doc}
+  "ezjsonm" {>= "1.3.0"}
+  "alcotest" {with-test}
+  "alcotest-lwt" {with-test}
+  "httpaf-lwt-unix" {with-test}
+  "cmdliner"
+  "http-lwt-client"
+]
```
Comment (mseri) [inline packages/equinoxe/equinoxe.0.1.0/opam:20]:
```suggestion
  "lwt" {>= "5.3.0"}
]
```

due to
```
ocamlc src/lib/.equinoxe.objs/byte/equinoxe__Http_api.{cmo,cmt} (exit 2)
# (cd _build/default && /home/opam/.opam/4.08/bin/ocamlc.opt -w -40 -g -bin-annot -I src/lib/.equinoxe.objs/byte -I /home/opam/.opam/4.08/lib/angstrom -I /home/opam/.opam/4.08/lib/asn1-combinators -I /home/opam/.opam/4.08/lib/astring -I /home/opam/.opam/4.08/lib/base/caml -I /home/opam/.opam/4.08/lib/base64 -I /home/opam/.opam/4.08/lib/bigarray-compat -I /home/opam/.opam/4.08/lib/bigstringaf -I /home/opam/.opam/4.08/lib/bos -I /home/opam/.opam/4.08/lib/bytes -I /home/opam/.opam/4.08/lib/ca-certs -I /home/opam/.opam/4.08/lib/cstruct -I /home/opam/.opam/4.08/lib/cstruct-sexp -I /home/opam/.opam/4.08/lib/dns -I /home/opam/.opam/4.08/lib/dns-client -I /home/opam/.opam/4.08/lib/dns-client/lwt -I /home/opam/.opam/4.08/lib/dns-client/resolvconf -I /home/opam/.opam/4.08/lib/dns/cache -I /home/opam/.opam/4.08/lib/domain-name -I /home/opam/.opam/4.08/lib/duration -I /home/opam/.opam/4.08/lib/eqaf -I /home/opam/.opam/4.08/lib/eqaf/bigstring -I /home/opam/.opam/4.08/lib/eqaf/cstruct -I /home/opam/.opam/4.08/lib/ezjsonm -I /home/opam/.opam/4.08/lib/faraday -I /home/opam/.opam/4.08/lib/faraday-lwt -I /home/opam/.opam/4.08/lib/faraday-lwt-unix -I /home/opam/.opam/4.08/lib/fmt -I /home/opam/.opam/4.08/lib/fpath -I /home/opam/.opam/4.08/lib/gmap -I /home/opam/.opam/4.08/lib/h2 -I /home/opam/.opam/4.08/lib/happy-eyeballs -I /home/opam/.opam/4.08/lib/happy-eyeballs-lwt -I /home/opam/.opam/4.08/lib/hex -I /home/opam/.opam/4.08/lib/hkdf -I /home/opam/.opam/4.08/lib/hpack -I /home/opam/.opam/4.08/lib/http-lwt-client -I /home/opam/.opam/4.08/lib/httpaf -I /home/opam/.opam/4.08/lib/ipaddr -I /home/opam/.opam/4.08/lib/ipaddr-sexp -I /home/opam/.opam/4.08/lib/ipaddr/unix -I /home/opam/.opam/4.08/lib/jsonm -I /home/opam/.opam/4.08/lib/logs -I /home/opam/.opam/4.08/lib/lru -I /home/opam/.opam/4.08/lib/lwt -I /home/opam/.opam/4.08/lib/lwt/unix -I /home/opam/.opam/4.08/lib/macaddr -I /home/opam/.opam/4.08/lib/metrics -I /home/opam/.opam/4.08/lib/mirage-crypto -I /home/opam/.opam/4.08/lib/mirage-crypto-ec -I /home/opam/.opam/4.08/lib/mirage-crypto-pk -I /home/opam/.opam/4.08/lib/mirage-crypto-rng -I /home/opam/.opam/4.08/lib/mirage-crypto-rng/lwt -I /home/opam/.opam/4.08/lib/mirage-crypto-rng/unix -I /home/opam/.opam/4.08/lib/mmap -I /home/opam/.opam/4.08/lib/mtime -I /home/opam/.opam/4.08/lib/mtime/os -I /home/opam/.opam/4.08/lib/ocaml/threads -I /home/opam/.opam/4.08/lib/parsexp -I /home/opam/.opam/4.08/lib/pbkdf -I /home/opam/.opam/4.08/lib/ppx_sexp_conv/runtime-lib -I /home/opam/.opam/4.08/lib/psq -I /home/opam/.opam/4.08/lib/ptime -I /home/opam/.opam/4.08/lib/ptime/os -I /home/opam/.opam/4.08/lib/randomconv -I /home/opam/.opam/4.08/lib/result -I /home/opam/.opam/4.08/lib/rresult -I /home/opam/.opam/4.08/lib/seq -I /home/opam/.opam/4.08/lib/sexplib -I /home/opam/.opam/4.08/lib/sexplib0 -I /home/opam/.opam/4.08/lib/stdlib-shims -I /home/opam/.opam/4.08/lib/tls -I /home/opam/.opam/4.08/lib/tls/lwt -I /home/opam/.opam/4.08/lib/uchar -I /home/opam/.opam/4.08/lib/uutf -I /home/opam/.opam/4.08/lib/x509 -I /home/opam/.opam/4.08/lib/zarith -intf-suffix .ml -no-alias-deps -open Equinoxe__ -o src/lib/.equinoxe.objs/byte/equinoxe__Http_api.cmo -c -impl src/lib/http_api.ml)
# File "src/lib/http_api.ml", line 26, characters 5-15:
# 26 | open Lwt.Syntax
#           ^^^^^^^^^^
# Error: Unbound module Lwt.Syntax
#     ocamlopt src/lib/.equinoxe.objs/native/equinoxe__Http_api.{cmx,o} (exit 2)
# (cd _build/default && /home/opam/.opam/4.08/bin/ocamlopt.opt -w -40 -g -I src/lib/.equinoxe.objs/byte -I src/lib/.equinoxe.objs/native -I /home/opam/.opam/4.08/lib/angstrom -I /home/opam/.opam/4.08/lib/asn1-combinators -I /home/opam/.opam/4.08/lib/astring -I /home/opam/.opam/4.08/lib/base/caml -I /home/opam/.opam/4.08/lib/base64 -I /home/opam/.opam/4.08/lib/bigarray-compat -I /home/opam/.opam/4.08/lib/bigstringaf -I /home/opam/.opam/4.08/lib/bos -I /home/opam/.opam/4.08/lib/bytes -I /home/opam/.opam/4.08/lib/ca-certs -I /home/opam/.opam/4.08/lib/cstruct -I /home/opam/.opam/4.08/lib/cstruct-sexp -I /home/opam/.opam/4.08/lib/dns -I /home/opam/.opam/4.08/lib/dns-client -I /home/opam/.opam/4.08/lib/dns-client/lwt -I /home/opam/.opam/4.08/lib/dns-client/resolvconf -I /home/opam/.opam/4.08/lib/dns/cache -I /home/opam/.opam/4.08/lib/domain-name -I /home/opam/.opam/4.08/lib/duration -I /home/opam/.opam/4.08/lib/eqaf -I /home/opam/.opam/4.08/lib/eqaf/bigstring -I /home/opam/.opam/4.08/lib/eqaf/cstruct -I /home/opam/.opam/4.08/lib/ezjsonm -I /home/opam/.opam/4.08/lib/faraday -I /home/opam/.opam/4.08/lib/faraday-lwt -I /home/opam/.opam/4.08/lib/faraday-lwt-unix -I /home/opam/.opam/4.08/lib/fmt -I /home/opam/.opam/4.08/lib/fpath -I /home/opam/.opam/4.08/lib/gmap -I /home/opam/.opam/4.08/lib/h2 -I /home/opam/.opam/4.08/lib/happy-eyeballs -I /home/opam/.opam/4.08/lib/happy-eyeballs-lwt -I /home/opam/.opam/4.08/lib/hex -I /home/opam/.opam/4.08/lib/hkdf -I /home/opam/.opam/4.08/lib/hpack -I /home/opam/.opam/4.08/lib/http-lwt-client -I /home/opam/.opam/4.08/lib/httpaf -I /home/opam/.opam/4.08/lib/ipaddr -I /home/opam/.opam/4.08/lib/ipaddr-sexp -I /home/opam/.opam/4.08/lib/ipaddr/unix -I /home/opam/.opam/4.08/lib/jsonm -I /home/opam/.opam/4.08/lib/logs -I /home/opam/.opam/4.08/lib/lru -I /home/opam/.opam/4.08/lib/lwt -I /home/opam/.opam/4.08/lib/lwt/unix -I /home/opam/.opam/4.08/lib/macaddr -I /home/opam/.opam/4.08/lib/metrics -I /home/opam/.opam/4.08/lib/mirage-crypto -I /home/opam/.opam/4.08/lib/mirage-crypto-ec -I /home/opam/.opam/4.08/lib/mirage-crypto-pk -I /home/opam/.opam/4.08/lib/mirage-crypto-rng -I /home/opam/.opam/4.08/lib/mirage-crypto-rng/lwt -I /home/opam/.opam/4.08/lib/mirage-crypto-rng/unix -I /home/opam/.opam/4.08/lib/mmap -I /home/opam/.opam/4.08/lib/mtime -I /home/opam/.opam/4.08/lib/mtime/os -I /home/opam/.opam/4.08/lib/ocaml/threads -I /home/opam/.opam/4.08/lib/parsexp -I /home/opam/.opam/4.08/lib/pbkdf -I /home/opam/.opam/4.08/lib/ppx_sexp_conv/runtime-lib -I /home/opam/.opam/4.08/lib/psq -I /home/opam/.opam/4.08/lib/ptime -I /home/opam/.opam/4.08/lib/ptime/os -I /home/opam/.opam/4.08/lib/randomconv -I /home/opam/.opam/4.08/lib/result -I /home/opam/.opam/4.08/lib/rresult -I /home/opam/.opam/4.08/lib/seq -I /home/opam/.opam/4.08/lib/sexplib -I /home/opam/.opam/4.08/lib/sexplib0 -I /home/opam/.opam/4.08/lib/stdlib-shims -I /home/opam/.opam/4.08/lib/tls -I /home/opam/.opam/4.08/lib/tls/lwt -I /home/opam/.opam/4.08/lib/uchar -I /home/opam/.opam/4.08/lib/uutf -I /home/opam/.opam/4.08/lib/x509 -I /home/opam/.opam/4.08/lib/zarith -intf-suffix .ml -no-alias-deps -open Equinoxe__ -o src/lib/.equinoxe.objs/native/equinoxe__Http_api.cmx -c -impl src/lib/http_api.ml)
# File "src/lib/http_api.ml", line 26, characters 5-15:
# 26 | open Lwt.Syntax
#           ^^^^^^^^^^
# Error: Unbound module Lwt.Syntax
```

File: packages/equinoxe/equinoxe.0.1.0/opam [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+build: [
```
Comment (mseri) [inline packages/equinoxe/equinoxe.0.1.0/opam:21]:
```suggestion
conflicts: [ "result" {< "1.5"} ]
build: [
```
To fix the remaining lower bounds

**Comment by mseri** [comment]
It is good now, please upstream the fixes above

**Comment by maiste** [comment]
Thanks, @mseri, for the correction! I upstream the fix you added.

**Pull request merged**